core/templates/services/context.service.spec.ts

```diff
@@ -25,15 +25,24 @@ import { UrlService } from 'services/contextual/url.service';
 describe('Context service', () => {
   let ecs: ContextService = null;
   let urlService: UrlService = null;
-
+  let resetContext = () => {
+    ContextService._pageContext = null;
+    ContextService._explorationIsLinkedToStory = false;
+    ContextService._explorationId = null;
+    ContextService._questionPlayerIsManuallySet = false;
+    ContextService._questionId = null;
+    ContextService._editorContext = null;
+    ContextService._customEntityContext = null;
+  };
+  afterEach(resetContext);
   describe('behavior in the exploration learner view', () => {
     beforeEach(() => {
       ecs = TestBed.get(ContextService);
       urlService = TestBed.get(UrlService);
       spyOn(urlService, 'getPathname').and.returnValue('/explore/123');
       spyOn(urlService, 'getHash').and.returnValue('');
     });
-
+    afterEach(resetContext);
```

Comment:
Hm, I think we need just one afterEach in each describe block. Can you check please?